cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -551,22 +551,37 @@ func parsePEMBlocks(raw []byte) (map[string]*pem.Block, error) {
 	return blocks, nil
 }
 
-func clusterHasInstance(opts GCPConfig, zone string, instanceID uint64) (bool, error) {
-	// zone looks like
+func clusterHasInstance(opts GCPConfig, instanceZone string, instanceID uint64) (bool, error) {
+	// instanceZone looks like
 	// "https://www.googleapis.com/compute/v1/projects/my-project/zones/us-central1-c"
 	// Convert it to just "us-central1-c".
-	zone = path.Base(zone)
+	instanceZone = path.Base(instanceZone)
+
 	recordMetric := csrmetrics.OutboundRPCStartRecorder("container.ProjectsZonesClustersService.Get")
-	cluster, err := container.NewProjectsZonesClustersService(opts.Container).Get(opts.ProjectID, zone, opts.ClusterName).Do()
+	cluster, err := container.NewProjectsZonesClustersService(opts.Container).Get(opts.ProjectID, opts.Location, opts.ClusterName).Do()
 	if err != nil {
 		recordMetric(csrmetrics.OutboundRPCStatusError)
 		return false, fmt.Errorf("fetching cluster info: %v", err)
 	}
 	recordMetric(csrmetrics.OutboundRPCStatusOK)
 	for _, np := range cluster.NodePools {
 		for _, ig := range np.InstanceGroupUrls {
-			igName := path.Base(ig)
-			ok, err := groupHasInstance(opts, zone, igName, instanceID)
+			igParts := strings.Split(ig, "/")
```

Comment:
Lines 569-574 are kind of difficult to parse what they are extracting. I think a helper func would be great here.

Reply:
done